sources/System/Health/HealthSystem.cpp

```
@@ -3,23 +3,70 @@
 #include "Component/Health/HealthComponent.hpp"
-#include "Component/Controllable/ControllableComponent.hpp"
+#include "Component/Bonus/BonusComponent.hpp"
 #include "Entity/Entity.hpp"
+#include "Items/Bonus.hpp"
+
+namespace RAY3D = RAY::Drawables::Drawables3D;
 
 namespace BBM
 {
 	HealthSystem::HealthSystem(WAL::Wal &wal)
 		: System(wal)
 	{}
 
-	void HealthSystem::onFixedUpdate(WAL::ViewEntity<HealthComponent> &entity)
+	void HealthSystem::_createBonus(Vector3f position, BonusComponent::BonusType bonusType, std::chrono::nanoseconds timer)
+	{
+		static std::map<BonusComponent::BonusType, std::string> map = {
+				{BonusComponent::BonusType::BOMBSTOCK, "assets/items/bombup"},
+				{BonusComponent::BonusType::SPEEDUP, "assets/items/speedup"},
+				//{BonusComponent::BonusType::EXPLOSIONINC, "assets/items/explosion"},
+				{BonusComponent::BonusType::DAMAGEINC, "assets/items/fireup"}
+		};
+		static std::vector<std::function<void (WAL::Entity &, const WAL::Entity &, CollisionComponent::CollidedAxis)>> func = {
+				&Bonus::BombUpBonus, &Bonus::SpeedUpBonus, //&Bonus::ExplosionRangeBonus,
+				&Bonus::DamageIncreasedBonus
+		};
+
+		if (bonusType == BonusComponent::BonusType::NOTHING)
+			return;
+		try {
+			this->_wal.scene->scheduleNewEntity("Bonus")
+				.addComponent<PositionComponent>(position)
+				.addComponent<HealthComponent>(1, [](WAL::Entity &entity) {
+					entity.scheduleDeletion();
+				})
+				.addComponent<LevitateComponent>(position.y)
+				.addComponent<CollisionComponent>([](WAL::Entity &bonus, const WAL::Entity &player, CollisionComponent::CollidedAxis axis) {
+					bonus.scheduleDeletion();
+				}, func[bonusType - 1], 0.25, .75)
+				.addComponent<TimerComponent>(timer, [](WAL::Entity &bonus, WAL::Wal &wal){
+					bonus.scheduleDeletion();
+				})
+				.addComponent<Drawable3DComponent, RAY3D::Model>(map.at(bonusType) + ".obj", std::make_pair(MAP_DIFFUSE, "assets/items/items.png"));
+		} catch (std::out_of_range const &err) {}
```

Don't use empty catch statements. You should never have to do that. In c++ you should check that you can do an action before doing it. Try/Catch are heavy and should not be used to silence errors.

This is the pythonic way of doing this but really not the C++ way.